packages/twenty-front/src/pages/settings/developers/webhooks/components/SettingsDevelopersWebhookDetail.tsx

```diff
@@ -72,30 +96,85 @@ export const SettingsDevelopersWebhooksDetail = () => {
 
   const isAnalyticsV2Enabled = useIsFeatureEnabled('IS_ANALYTICS_V2_ENABLED');
 
-  const fieldTypeOptions = [
-    { value: '*', label: 'All Objects' },
-    ...objectMetadataItems.map((item) => ({
-      value: item.nameSingular,
-      label: item.labelSingular,
-    })),
+  const fieldTypeOptions: SelectOption<string>[] = useMemo(
+    () => [
+      { value: '*', label: 'All Objects', Icon: IconNorthStar },
+      ...objectMetadataItems.map((item) => ({
+        value: item.nameSingular,
+        label: item.labelSingular,
+        Icon: getIcon(item.icon),
+      })),
+    ],
+    [objectMetadataItems, getIcon],
+  );
+
+  const actionOptions: SelectOption<string>[] = [
+    { value: '*', label: 'All Actions', Icon: IconNorthStar },
+    { value: 'create', label: 'Create', Icon: IconPlus },
+    { value: 'update', label: 'Update', Icon: IconRefresh },
+    { value: 'delete', label: 'Delete', Icon: IconTrash },
   ];
 
   const { updateOneRecord } = useUpdateOneRecord<Webhook>({
     objectNameSingular: CoreObjectNameSingular.Webhook,
   });
 
+  const cleanAndFormatOperations = (operations: WebhookOperationType[]) => {
+    return Array.from(
+      new Set(
+        operations
+          .filter((op) => isDefined(op.object) && isDefined(op.action))
+          .map((op) => `${op.object}.${op.action}`),
+      ),
+    );
+  };
+
   const handleSave = async () => {
+    const cleanedOperations = cleanAndFormatOperations(operations);
     setIsDirty(false);
     await updateOneRecord({
       idToUpdate: webhookId,
       updateOneRecordInput: {
-        operation: `${operationObjectSingularName}.${operationAction}`,
+        operations: cleanedOperations,
         description: description,
       },
     });
     navigate(developerPath);
   };
 
+  const updateOperation = (
+    index: number,
+    field: 'object' | 'action',
+    value: string | null,
+  ) => {
+    const newOperations = [...operations];
+    newOperations[index] = {
+      ...newOperations[index],
+      [field]: value ?? '',
+    };
+    setOperations(newOperations);
+    setIsDirty(true);
+
+    if (
+      index === operations.length - 1 &&
+      isDefined(value) &&
+      newOperations[index].object !== null &&
+      newOperations[index].action !== null &&
+      (newOperations[index].object !== '*' ||
+        newOperations[index].action !== '*')
+    ) {
+      setOperations([...newOperations, WEBHOOK_EMPTY_OPERATION]);
+    }
```

Comment:
I would do something like this: if no `*.*` operation and no operation with null action or null object, then add new empty operation
```suggestion
if (
  {object: "*", action: "*"} not in newOperations &&
newOperations.filter((operation) => operation.object === null || operation.action === null).length === 0
) {
  setOperations([...newOperations, WEBHOOK_EMPTY_OPERATION]);
}
```